alpine-infra/src/main/java/alpine/persistence/AlpineQueryManager.java

```diff
@@ -98,9 +105,10 @@ public AlpineQueryManager(final PersistenceManager pm, final AlpineRequest reque
      */
     public ApiKey getApiKey(final String key) {
         return callInTransaction(() -> {
-            final Query<ApiKey> query = pm.newQuery(ApiKey.class, "key == :key");
-            query.setParameters(key);
-            return executeAndCloseUnique(query);
+            final Query<ApiKey> query = pm.newQuery(ApiKey.class, "suffix == :suffix");
+            query.setParameters(key.substring(key.length() - SUFFIX_LENGTH));
+            ApiKey apiKey = executeAndCloseUnique(query);
+            return BCrypt.checkpw(key.substring(0, key.length() - SUFFIX_LENGTH), apiKey.getKey()) ? apiKey : null;
```

Comment:
Keep in mind that `apiKey` can be null if no matching record was found in the database.